CommunityToolkit.Mvvm/Messaging/WeakReferenceMessenger.cs

```diff
@@ -218,6 +218,12 @@ public void UnregisterAll(object recipient)
     public void UnregisterAll<TToken>(object recipient, TToken token)
         where TToken : IEquatable<TToken>
     {
+        // This method is never called with the unit type
+        if (typeof(TToken) == typeof(Unit))
+        {
+            throw new NotImplementedException();
```

Comment:
Since `Unit` is internal, what's the reason for throwing? Comment doesn't make it clear
Also Is `NotImplementedException` the best exception to use here?

Reply:
The reason is just that the rest of the method is not valid if `Unit` was used, so we might as well just have an explicit check to ensure that it is indeed never called. Since `Unit` is a value type, this type check will be compeltely removed by the compiler anyway when not used, so it will be completely free. There is no check being done at runtime here 🙂

Reply:
Ahh so it'll never actually throw NotImplementedException at runtime. I assumed we avoid throwing this in a production app -- maybe we should change it to InvalidOperationException or NotSupportedException to set a good example?

Reply:
I think `NotImplementedException` is fine here. The exact exception type doesn't really matter much anyway, this is not an actual exception we'll ever be throwing. If that had been the case, then yeah I'd have opted for `NotSupportedException`. But in this case this path is actually, by design, not implemented. I've added more documentation in 17b046513ca51c093ccbc0b58f078011a6072a5a though, which should hopefully make the setup clearer and easier to understand, let me know if that works! 🙂